cpp/tools/parquet/parquet_dump_footer.cc

```diff
@@ -0,0 +1,248 @@
+  }
+}
+
+// Replace the contents of s with random data of the same length.
+void Scrub(std::string* s) {
+  static char pool[4096];
+  static std::mt19937 rng(std::random_device{}());
+  static const bool kPoolInit = [] {
+    std::uniform_int_distribution<> caps(65, 90);
+    for (size_t i = 0; i < sizeof(pool); i++) pool[i] = caps(rng);
+    return true;
+  }();
+  (void)kPoolInit;
+
+  const size_t n = s->size();
+  s->clear();
+  while (s->size() < n) {
+    // To avoid repeating patterns we start somewhere up to halfway through the pool and
+    // append up to half the pool chars.
+    std::uniform_int_distribution<size_t> half(0, sizeof(pool) / 2 - 1);
+    size_t m = std::min(n, half(rng) + 1 /* at least one */);
+    s->append(&pool[half(rng)], m);
+  }
+}
+
+void Scrub(parquet::format::FileMetaData* md) {
+  for (auto& s : md->schema) {
+    Scrub(&s.name);
+  }
+  for (auto& r : md->row_groups) {
+    for (auto& c : r.columns) {
+      Scrub(&c.file_path);
+      if (c.__isset.meta_data) {
+        auto& m = c.meta_data;
+        for (auto& p : m.path_in_schema) Scrub(&p);
+        for (auto& kv : m.key_value_metadata) {
+          Scrub(&kv.key);
+          Scrub(&kv.value);
+        }
+        Scrub(&m.statistics.max_value);
+        Scrub(&m.statistics.min_value);
+        Scrub(&m.statistics.min);
+        Scrub(&m.statistics.max);
+      }
+
+      if (c.crypto_metadata.__isset.ENCRYPTION_WITH_COLUMN_KEY) {
+        auto& m = c.crypto_metadata.ENCRYPTION_WITH_COLUMN_KEY;
+        for (auto& p : m.path_in_schema) Scrub(&p);
+        Scrub(&m.key_metadata);
+      }
+      Scrub(&c.encrypted_column_metadata);
+    }
+  }
+  for (auto& kv : md->key_value_metadata) {
+    Scrub(&kv.key);
+    Scrub(&kv.value);
+  }
+  Scrub(&md->footer_signing_key_metadata);
+}
+
+// Returns:
+// - 0 on success
+// - -1 on error
+// - the size of the footer if tail is too small
+int64_t ParseFooter(const std::string& tail, parquet::format::FileMetaData* md) {
+  if (tail.size() > std::numeric_limits<int32_t>::max()) return -1;
+
+  const char* p = tail.data();
+  const int32_t n = static_cast<int32_t>(tail.size());
+  int32_t len = ReadLE32(p + n - 8);
+  if (len > n - 8) return len;
+
+  if (!Deserialize(tail.data() + n - 8 - len, len, md)) return -1;
+  return 0;
+}
+}  // namespace
+
+int main(int argc, char** argv) {
+  bool help = false;
+  bool scrub = true;
+  bool json = false;
+  std::string in;
+  std::string out;
+  for (int i = 1; i < argc; i++) {
+    char* arg = argv[i];
+    help |= !std::strcmp(arg, "-h") || !std::strcmp(arg, "--help");
+    scrub &= !!std::strcmp(arg, "--no-scrub");
+    json |= !std::strcmp(arg, "--json");
+    if (!std::strcmp(arg, "--in")) {
+      if (i + 1 >= argc) return PrintHelp();
+      in = argv[++i];
+    }
+    if (!std::strcmp(arg, "--out")) {
+      if (i + 1 >= argc) return PrintHelp();
+      out = argv[++i];
+    }
+  }
+  if (help || in.empty()) return PrintHelp();
+  std::string path;
```

Comment:
I would recommend to keep `main` to the minimum (for example only command-line option parsing), then delegate actual functionality to helper function(s).

Reply:
Done.